core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSail.java

```diff
@@ -863,4 +884,79 @@ public void setDashDataShapes(boolean dashDataShapes) {
 	}
+
+	@Override
+	public Map<IRI, Literal> getSettings() {
+		Map<IRI, Literal> settings = super.getSettings();
+
+		settings.put(PARALLEL_VALIDATION, BooleanLiteral.valueOf(isParallelValidation()));
+		settings.put(UNDEFINED_TARGET_VALIDATES_ALL_SUBJECTS,
+				BooleanLiteral.valueOf(isUndefinedTargetValidatesAllSubjects()));
+		settings.put(LOG_VALIDATION_PLANS, BooleanLiteral.valueOf(isLogValidationPlans()));
+		settings.put(LOG_VALIDATION_VIOLATIONS, BooleanLiteral.valueOf(isLogValidationViolations()));
+		settings.put(IGNORE_NO_SHAPES_LOADED_EXCEPTION, BooleanLiteral.valueOf(isIgnoreNoShapesLoadedException()));
+		settings.put(VALIDATION_ENABLED, BooleanLiteral.valueOf(isValidationEnabled()));
+		settings.put(CACHE_SELECT_NODES, BooleanLiteral.valueOf(isCacheSelectNodes()));
+		settings.put(GLOBAL_LOG_VALIDATION_EXECUTION, BooleanLiteral.valueOf(isGlobalLogValidationExecution()));
+		settings.put(RDFS_SUB_CLASS_REASONING, BooleanLiteral.valueOf(isRdfsSubClassReasoning()));
+		settings.put(PERFORMANCE_LOGGING, BooleanLiteral.valueOf(isPerformanceLogging()));
+		settings.put(SERIALIZABLE_VALIDATION, BooleanLiteral.valueOf(isSerializableValidation()));
+		settings.put(SHACL_ADVANCED_FEATURES, BooleanLiteral.valueOf(isShaclAdvancedFeatures()));
+		settings.put(DASH_DATA_SHAPES, BooleanLiteral.valueOf(isDashDataShapes()));
+
+		return settings;
+	}
+
+	@Override
+	public void setSettings(Map<IRI, Literal> settings) {
+		super.setSettings(settings);
+
+		settings.forEach((key, value) -> {
+
+			if (key.equals(PARALLEL_VALIDATION)) {
+				setParallelValidation(value.booleanValue());
+			}
+			if (key.equals(UNDEFINED_TARGET_VALIDATES_ALL_SUBJECTS)) {
+				setUndefinedTargetValidatesAllSubjects(value.booleanValue());
+			}
+			if (key.equals(LOG_VALIDATION_PLANS)) {
+				setLogValidationPlans(value.booleanValue());
+			}
+			if (key.equals(LOG_VALIDATION_VIOLATIONS)) {
+				setLogValidationViolations(value.booleanValue());
+			}
+			if (key.equals(IGNORE_NO_SHAPES_LOADED_EXCEPTION)) {
+				setIgnoreNoShapesLoadedException(value.booleanValue());
+			}
+			if (key.equals(VALIDATION_ENABLED)) {
+				if (value.booleanValue()) {
+					enableValidation();
+				} else {
+					disableValidation();
+				}
+			}
+			if (key.equals(CACHE_SELECT_NODES)) {
+				setCacheSelectNodes(value.booleanValue());
+			}
+			if (key.equals(GLOBAL_LOG_VALIDATION_EXECUTION)) {
+				setGlobalLogValidationExecution(value.booleanValue());
+			}
+			if (key.equals(RDFS_SUB_CLASS_REASONING)) {
+				setRdfsSubClassReasoning(value.booleanValue());
+			}
+			if (key.equals(PERFORMANCE_LOGGING)) {
+				setPerformanceLogging(value.booleanValue());
+			}
+			if (key.equals(SERIALIZABLE_VALIDATION)) {
+				setSerializableValidation(value.booleanValue());
+			}
+			if (key.equals(SHACL_ADVANCED_FEATURES)) {
+				setShaclAdvancedFeatures(value.booleanValue());
+			}
+			if (key.equals(DASH_DATA_SHAPES)) {
+				setDashDataShapes(value.booleanValue());
+			}
+
+		});
+	}
```

Comment:
Here is the ShailSail implementation aspect of the new settings approach. Looking at this I'm partly wondering if the old fashioned model based sail config could just delegate to these settings methods.

Also an important note. For settings that are unchangeable once the sail has been initialised we would either ignore them or throw an exception.